src/NServiceBus.Core/Persistence/InMemory/SagaPersister/InMemorySagaPersister.cs

```diff
@@ -160,7 +168,41 @@ static IContainSagaData DeepClone(IContainSagaData source)
 
             public IContainSagaData GetSagaCopy()
             {
-                return DeepClone(data);
+                var canBeClonedShallowly = canBeClonedShallowlyCache.GetOrAdd(data.GetType(), CanBeMemberwiselyCloned);
```

Comment:
Also, changing the second argument of the `GetOrAdd` call to a func, `type => CanBeMemberwiselyCloned(type)`, means we'll save a delegate allocation per call here.